tools/gather-sysinfo/gather-sysinfo.go

```diff
@@ -1,79 +1,108 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"io/ioutil"
-	"log"
 	"os"
 	"path/filepath"
 	"time"
 
 	"github.com/jaypipes/ghw/pkg/snapshot"
+	"github.com/openshift-kni/debug-tools/pkg/knit/cmd"
+	"github.com/spf13/cobra"
 )
 
+type snapshotOptions struct {
+	dumpList  bool
+	output    string
+	rootDir   string
+	sleepTime int
+}
+
 func main() {
-	debug := flag.Bool("debug", false, "enable debug output")
-	output := flag.String("output", "", "path to clone system information into")
-	rootDir := flag.String("root", "", "pseudofs root - use this if running inside a container")
-	dumpList := flag.Bool("dump", false, "just dump the glob list of expected content and exit")
+	root := cmd.NewRootCommand(newSnapshotCommand)
+
+	if err := root.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
```

Comment:
Are you sure we need to print the err? I think cobra already prints the error if Execute fails and we will end up with the same error twice: At the beginning and after the help output, with little/no difference between them. Is that the desired behavior?

Reply:
good point, will check

Reply:
current code:
```
$ ./build/_output/bin/gather-sysinfo irqaff -X
unknown shorthand flag: 'X' in -X
$ ./build/_output/bin/gather-sysinfo foo
unknown command "foo" for "knit"
$ ./build/_output/bin/gather-sysinfo
Usage:
  knit [flags]
  knit [command]

Available Commands:
  cpuaff      show cpu thread affinities
  help        Help about any command
  irqaff      show IRQ/softirq thread affinities
  podres      show currently allocated pod resources
  snapshot    snapshot pseudofilesystems for offline analysis
  wait        wait forever, or until a UNIX signal (SIGINT, SIGTERM) arrives

Flags:
  -C, --cpulist string   isolated cpu set to check (see man (7) cpuset - List format (default "0-16383")
  -D, --debug            enable debug log
  -h, --help             help for knit
  -J, --json             output as JSON
  -P, --procfs string    procfs root (default "/proc")
  -S, --sysfs string     sysfs root (default "/sys")

Use "knit [command] --help" for more information about a command.
```
without the line above:
```
$ ./build/_output/bin/gather-sysinfo irqaff -X
1060 13:20:33 fromani@musashi2 ~/Projects/golang/src/github.com/openshift-kni/performance-addon-operators $ echo $?
1
$ ./build/_output/bin/gather-sysinfo foo
$ echo $?
1
$ ./build/_output/bin/gather-sysinfo
Usage:
  knit [flags]
  knit [command]

Available Commands:
  cpuaff      show cpu thread affinities
  help        Help about any command
  irqaff      show IRQ/softirq thread affinities
  podres      show currently allocated pod resources
  snapshot    snapshot pseudofilesystems for offline analysis
  wait        wait forever, or until a UNIX signal (SIGINT, SIGTERM) arrives

Flags:
  -C, --cpulist string   isolated cpu set to check (see man (7) cpuset - List format (default "0-16383")
  -D, --debug            enable debug log
  -h, --help             help for knit
  -J, --json             output as JSON
  -P, --procfs string    procfs root (default "/proc")
  -S, --sysfs string     sysfs root (default "/sys")

Use "knit [command] --help" for more information about a command.
```

Reply:
So I'd say let's keep it.

Reply:
for sure... thanks. Weird, but we have  a different behavior in PPC.
/lgtm